exporters/elasticsearch/src/es_log_exporter.cc

```diff
@@ -73,6 +76,23 @@ class ResponseHandler : public http_client::EventHandler
   // Callback method when an http event occurs
   void OnEvent(http_client::SessionState state, nostd::string_view reason) noexcept override
   {
+    // need to modify stop_waiting_ under lock before calling notify_all
+    switch (state)
+    {
+      case http_client::SessionState::CreateFailed:
+      case http_client::SessionState::ConnectFailed:
```

Comment:
Would it be possible to add the enumeration values which are not handled?
[enumeration value 'CreateFailed' not handled in switch [-Wswitch]](https://github.com/open-telemetry/opentelemetry-cpp/runs/4694407563?check_suite_focus=true#step:5:117)